src/microbe_stage/systems/SiderophoreSystem.cs

```diff
@@ -0,0 +1,137 @@
+﻿namespace Systems;
+
+using System;
+using Components;
+using DefaultEcs;
+using DefaultEcs.System;
+using DefaultEcs.Threading;
+using World = DefaultEcs.World;
+
+/// <summary>
+///   Handles siderophore projectile collisions
+/// </summary>
+[With(typeof(CollisionManagement))]
+[With(typeof(Physics))]
+[With(typeof(TimedLife))]
+[With(typeof(SiderophoreProjectile))]
+[RunsAfter(typeof(PhysicsCollisionManagementSystem))]
+[RuntimeCost(0.5f, false)]
+public sealed class SiderophoreSystem : AEntitySetSystem<float>
+{
+    /// <summary>
+    ///   Holds a persistent instance of the collision filter callback to not need to create multiple delegates, and
+    ///   to make doubly sure this callback won't be garbage collected while the native side still has a reference to
+    ///   it.
+    /// </summary>
+    private readonly IWorldSimulation worldSimulation;
+
+    public SiderophoreSystem(World world, IParallelRunner runner, IWorldSimulation worldSimulation) :
+        base(world, runner)
+    {
+        this.worldSimulation = worldSimulation;
+    }
+
+    protected override void Update(float delta, in Entity entity)
+    {
+        if (!entity.Has<SiderophoreProjectile>())
```

Comment:
This is not required as the With attribute requires this system can only run on entities that have this component.